src/warnet/server.py

```diff
@@ -376,13 +389,15 @@ def network_from_file(
         Run a warnet with topology loaded from a <graph_file>
         """
 
-        def thread_start(wn, lock: threading.Lock):
-            with lock:
+        def thread_start(server: Server, network):
+            with server.image_build_lock:
                 try:
+                    wn = server.get_warnet(network)
                     wn.generate_deployment()
-                    # wn.write_fork_observer_config()
+                    wn.write_fork_observer_config()
```

Comment:
in "server: optimistic warnet cache on server" (https://github.com/bitcoin-dev-project/warnet/pull/254/commits/be145a400d0a4ed43685558d005a1f93eb1f8937):

did you mean to add fork observer back in here? this whole commit seems to be unrelated to the IP changes, might be safer to drop it for now so we can merge just getting rid of the IP code?

Reply:
Yeah I see why it's not obvious, sorry my bad.

Current behaviour is that we rebuild the warnet from deployment file on *every* command run against the server. This doesn't work when we remove the hardcoded IP addresses because we are now retrieving them from the backend. This meant that every single server RPC would cause lookup of every ip from the backend on every call. Yuck.

To combat this I attempted the warnet caching, alongside the ip address caching in the previous comment. I think these are still a bit naive though, as you point out. I am trying to work out how we can totally switch to hostnames, which I think should be doable in most cases, and then we will neither need to cache or lookup ips all the time...

Reply:
It actually looks like it could work quite nicely by only modifying lnnode.py ... which would be nice